Docs: Split Subprojects in Explanation and How-to (Diátaxis) (#9785)
readthedocs/readthedocs.org: docs/user/guides/subprojects.rst

How To Add a Subproject
=======================

This guide shows you how to add a Subproject to your main project.

.. seealso::

   :doc:`/subprojects`
      Read more about what the subproject feature can do and how it works.

Adding a subproject
-------------------

> In the admin dashboard for your project, select "Subprojects" from the menu.

Reviewer: ```suggestion
In the admin dashboard for your project, select :guilabel:`Subprojects` from the menu.
```

When we talk about elements on the UI we have been using `:guilabel:`

> From this page you can add a subproject by typing in the project slug.

Author: ```suggestion
From this page you can add a subproject by typing in the project slug.

.. note::

   * On |org_brand|, you need to be maintainer or owner of a subproject in order to choose it from your main project.
   * On |com_brand|, you need to be Owner or have write access to the subproject in order to choose it from your main project.
```

Author: Is this accurate? CC: @humitos @ericholscher ?

Reviewer 2: Write access isn't really a concept on .com. I think you mean admin permissions?

Author: Updated the suggestion and added it :+1:

Subproject aliases
~~~~~~~~~~~~~~~~~~

> You can use an alias for the subproject when it is created. This allows you to override the URL that is used to access it, giving more configurability to how you want to structure your projects.

Author: Unresolving because I have a question :)

> You can use an alias for the subproject when it is created. This allows you to override the URL that is used to access it, giving more configurability to how you want to structure your projects.

Reviewer: Yeah, I think your interpretation is fine.

Subprojects also have an impact in how search works. You can find in all the subprojects easily. I'm not sure if we are mentioning this on this page too --but I just realize that while reading your "how it works" :)

> You can use an alias for the subproject when it is created. This allows you to override the URL that is used to access it, giving more configurability to how you want to structure your projects.

Author: Very good point re:search, will mention that :+1:

> You can use an alias for the subproject when it is created. This allows you to override the URL that is used to access it, giving more configurability to how you want to structure your projects.

Author: * There is now an example for aliases in the How-to :+1: 
* Common search is mentioned in the introduction for the subproject explanation article :+1:

> How To Add a Subproject

Reviewer 2: Seems this should be `Manage` a project, with a section on adding, removing, and aliases?

=======================

This guide shows you how to add a subproject to your main project.

.. seealso::

   :doc:`/subprojects`
      Read more about what the subproject feature can do and how it works.

Adding a subproject
-------------------

In the admin dashboard for your project, select :guilabel:`Subprojects` from the menu.
From this page you can add a subproject by typing in the project slug.

Subproject aliases
~~~~~~~~~~~~~~~~~~

You can choose an alias for the subproject when it is created.
This allows you to override the URL that is used to access it,
giving more control over how you want to structure your projects.

> You can set your subproject's project name and slug however you want,
> but we suggest prefixing it with the name of the main project.

Reviewer 2: ```suggestion
You can set your subproject's project name and :term:`slug` however you want,
but we suggest prefixing it with the name of the main project.
```

We say URL above, and slug here. I'm not sure what this sentence is saying though? 

I think it's generally best to avoid using `slug`, since it's definitely jargon.

Typically, a subproject is created with a ``<mainproject>-`` prefix,
for instance if the main project is called ``foo`` and the subproject is called ``bar``,
then the subproject's Read the Docs project name will be ``foo-bar``.
When adding the subproject,

> the alias is set to ``bar`` and the project's URL becomes

Reviewer 2: I think this is saying the same thing as the paragraph above, but in a much clearer way.

Author: Definitely! And all of this was explanation so I've removed it from the How-to :)

How to manage subprojects
=========================

This guide shows you how to manage subprojects,
which are a way to host multiple projects under a "main project".

.. seealso::

   :doc:`/subprojects`
      Read more about what the subproject feature can do and how it works.

Adding a subproject
-------------------

> In the admin dashboard for your project, select :guilabel:`Subprojects` from the menu.

Reviewer 2: We definitely need a better name for this than `left menu`.

From this page you can add a subproject by choosing the subproject in the :guilabel:`Subproject` dropdown

> and typing an alias in the :guilabel:`Alias` field.

Author: This is definitely better. There was a lot of "subproject" in that sentence.

Immediately after adding the subproject, it will be visible from the link displayed in the updated list of subprojects.

.. image:: /img/screenshot_subprojects_list.png
    :alt: Screenshot of a subproject immediately visible in the list after creation

> .. note::
> 
>    * On |org_brand|, you need to be maintainer of a subproject in order to choose it from your main project.
>    * On |com_brand|, you need to have admin access to the subproject in order to choose it from your main project.

Reviewer: I propose to not use bullet list for this differentiation. We can use the following approach, which I think it renders better and it's easier to read:

```suggestion
.. note::

   |org_brand|
      You need to be *maintainer* of a subproject in order to choose it from your main project.

   |com_brand|
      You need to have *admin access* to the subproject in order to choose it from your main project.
```

### Example

![Screenshot_2022-12-21_17-08-34](https://user-images.githubusercontent.com/244656/208950421-79a54ffb-e589-41bb-b20d-d5535efe9cbb.png)

Editing a subproject
--------------------

You can edit a subproject at any time by clicking :guilabel:`📝️` in the list of subprojects.
On the following page, it's possible to both change the subproject and its alias
using the :guilabel:`Subproject` dropdown and the :guilabel:`Alias` field.
Click :guilabel:`Update subproject` to save your changes.

> The documentations served at``/projects/<subproject-alias>/`` will be updated immediately when you save your changes.

Reviewer: ```suggestion
The documentations served at ``/projects/<subproject-alias>/`` will be updated immediately when you save your changes.
```

Author: Oh nicely spotted!! :eagle: :mag_right:

Deleting a subproject
---------------------

You can delete a subproject at any time by clicking :guilabel:`📝️` in the list of subprojects.
On the edit page, click :guilabel:`Delete subproject`.

Your subproject will be removed immediately and will be served from it's own domain:

> * Previously it was served at: `<source-project-domain>/projects/<subproject-alias>/`

Reviewer: ```suggestion
* Previously it was served at: `<main-project-domain>/projects/<subproject-alias>/`
```

I think we are not using "source project" for this.

Author: Nope, we definitely want main project, so we can have a more consistent terminology.

* Now it will be served at `<subproject-domain>/`

> **Deleting a subproject does not mean to completely remove that project**.

Reviewer: ```suggestion
Deleting a subproject does not mean to completely remove that project.
```

I would not emphasize a negated sentence, since it communicates a warning or the inability to undo the step.

Reviewer: A rephrase may work better for emphasize:

```rst
**Deleting a subproject only removes the reference from the main project.**
It does not completely remove that project.
```

> **Deleting a subproject does not mean to completely remove that project**.
> It only removes the reference from the main project.

Reviewer: ```suggestion
**Deleting a subproject only removes the reference from the main project.**
It does not completely remove that project.
```

My previous comment as a suggestion 😄

Author: Yes, it's an improvement to have the emphasized text emphasize something that happens and not something that doesn't happen :+1:

How to manage subprojects
=========================

This guide shows you how to manage subprojects,
which is a way to host multiple projects under a "main project".

.. seealso::

   :doc:`/subprojects`
      Read more about what the subproject feature can do and how it works.

Adding a subproject
-------------------

In the admin dashboard for your project, select :guilabel:`Subprojects` from the left menu.
From this page you can add a subproject by choosing a project from the :guilabel:`Subproject` dropdown
and typing an alias in the :guilabel:`Alias` field.

Immediately after adding the subproject, it will be hosted at the URL displayed in the updated list of subprojects.

> .. image:: /img/screenshot_subprojects_list.png

Reviewer 2: Re: @humitos comment about HowTo's, screenshots also don't match across .com & .org currently. It will be better once we ship the new templates.

Author: Ah yeah, maybe with tabbed panels, we can find a clever way to fix this in the future.

It's okay to stick with .org screenshots for now, right? Pretty sure that's the general pattern?

Reviewer 2: Yea, it's fine.

In the admin dashboard for your project, select :guilabel:`Subprojects` from the menu.
From this page you can add a subproject by typing in the project slug.

> Immediately after adding the subproject, its default version will be visible from the link displayed in the list of subprojects.

Author: Screenshots usually make things better and in this case, it also did :)

.. note::

   * On |org_brand|, you need to be maintainer of a subproject in order to choose it from your main project.
   * On |com_brand|, you need to have admin access to the subproject in order to choose it from your main project.

Editing a subproject
--------------------

> You can edit a subproject at any time by clicking :guilabel:`|:memo:|` in the list of subprojects.

Reviewer 2: <img width="714" alt="Screenshot 2022-12-16 at 10 06 39 AM" src="https://user-images.githubusercontent.com/25510/208161232-e3c236d8-32f4-44dc-96f3-2de05487464b.png">
Also it didn't render properly :)

> You can edit a subproject at any time by clicking :guilabel:`|:memo:|` in the list of subprojects.

Author: I like the direct use of emojis much more, my edit (GEdit :scream: ) supports it very well.

Since it doesn't render properly, I think we have to consider just using emojis directly.

Also, that button should have more than an icon, it's not accessibility-friendly, which makes it not-so documentation friendly at the same time.

> You can edit a subproject at any time by clicking :guilabel:`|:memo:|` in the list of subprojects.

Author: Using a real emoji now! Incidentally, this was when I found out that my PyCharm doesn't even display emojis :upside_down_face:

On the following page, it's possible to both change the alias and source project.

> Click :guilabel:`Update subproject` to save your changes.

Author: I now remember that I skipped these to avoid using the `Child` label since it broke the terminology. Will fix that in a separate PR.

> What is visible in ``/projects/<subproject-alias>/`` will be updated immediately when you save your changes.

Author: Yes exactly, no re-builds necessary etc.

Deleting a subproject
---------------------

You can edit a subproject at any time by clicking :guilabel:`|:memo:|` in the list of subprojects.
On the edit page, click :guilabel:`Delete subproject`.

> Your subproject will be removed immediately and it will stop redirecting from its own domain.

Author: Yes, "placeholders" really need a standard, it should be a common problem for all technical writers on the globe :)

> Deleting a subproject does not mean to completely remove that project.
> It is solely about removing the reference from the main project.

Reviewer 2: ```suggestion
**Deleting a subproject does not mean to completely remove that project**.
It removes the reference from the main project.
```

Perhaps this is useful. I think an admonition is probably too much here, but it's good to highlight.
